### PR TITLE
Fixed problem with Corr numberpicker when using Insightpump

### PR DESCRIPTION
**Pull request by harispoljo**
Can @jamorham check?

### CODE REVIEW

**Comment by harispoljo** [comment]
Fixed #1196

**Comment by MilosKozak** [comment]
@TebbeUbben ok?

File: app/src/main/java/info/nightscout/androidaps/plugins/PumpInsight/InsightPlugin.java [hunk 1]
```diff
@@ -914,7 +914,6 @@ private PumpEnactResult pumpEnactFailure() {
     public Constraint<Double> applyBolusConstraints(Constraint<Double> insulin) {
         if (statusResult != null) {
             insulin.setIfSmaller(statusResult.maximumBolusAmount, String.format(MainApp.gs(R.string.limitingbolus), statusResult.maximumBolusAmount, MainApp.gs(R.string.pumplimit)), this);
-            insulin.setIfGreater(statusResult.minimumBolusAmount, String.format(MainApp.gs(R.string.limitingbolus), statusResult.maximumBolusAmount, MainApp.gs(R.string.pumplimit)), this);
```
Comment (TebbeUbben) [inline app/src/main/java/info/nightscout/androidaps/plugins/PumpInsight/InsightPlugin.java]: You should leave that line, but skip it if the insulin amount is 0.

Comment (harispoljo) [reply app/src/main/java/info/nightscout/androidaps/plugins/PumpInsight/InsightPlugin.java]: I booted it up on my rig and it still didn't work however I noticed that it only trigger for values between 0 and 0.04

File: app/src/main/java/info/nightscout/androidaps/plugins/PumpInsight/InsightPlugin.java [hunk 2]
```diff
@@ -914,7 +914,9 @@ private PumpEnactResult pumpEnactFailure() {
...
+
+            if(insulin.value() < 0 && insulin.value() > 0.04 )
```
Comment (jotomo) [inline app/src/main/java/info/nightscout/androidaps/plugins/PumpInsight/InsightPlugin.java]: This condition is always false. _value_ can't be negative AND greater 0.04 at the same time.
If this is to exclude an amount the pump can't deliver, then I think that's worth a comment or constant to help the maintainer of the code understand why this is here.

**Comment by harispoljo** [comment]
Wait why did this get merged?

The problem with the numberpicker is that it increaments with 0.01 so only way to get past 0.04 is to 
type in the value 0.05 instead of pressing + because that would trigger "Bolus constrains applied". I dont understand why there is a check for the numberpicker when there always will be a check if the amount of insulin after you click ok.

**Comment by TebbeUbben** [comment]
Okay, then you need to fix the number picker. The insight plugin is fine now. What you have fixed is the issue that AAPS wanted to deliver 0.05 even if you didn't enter any insulin in the treatment dialog. Trying to deliver less than the minimum amount (which is dynamically read from the pump) will cause it to throw an error.

**Pull request merged**